apps/web/app/not-found.tsx

```diff
@@ -20,7 +20,22 @@ enum PageType {
 function getPageInfo(pathname: string, host: string) {
   const { isValidOrgDomain, currentOrgDomain } = getOrgDomainConfigFromHostname({ hostname: host });
   const [routerUsername] = pathname?.replace("%20", "-").split(/[?#]/) ?? [];
-  if (!routerUsername || (isValidOrgDomain && currentOrgDomain)) {
+  if (routerUsername && (!isValidOrgDomain || !currentOrgDomain)) {
+    const splitPath = routerUsername.split("/");
+    if (splitPath[1] === "team" && splitPath.length === 3) {
+      return {
+        username: splitPath[2],
+        pageType: PageType.TEAM,
+        url: `${WEBSITE_URL}/signup?callbackUrl=settings/teams/new%3Fslug%3D${splitPath[2].replace("/", "")}`,
+      };
+    } else {
+      return {
+        username: routerUsername,
+        pageType: PageType.USER,
+        url: `${WEBSITE_URL}/signup?username=${routerUsername.replace("/", "")}`,
+      };
+    }
+  } else {
```

Comment:
Noticed that logic isn't exactly the same as in pages/404, so I made it same. safer that way.